torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -45,21 +48,43 @@ static IValue Table(
   return Tup(std::move(ivalue_entries));
 }
 
-c10::IValue getFunctionTuple(const Function& func) {
+std::string getModulePath(Node* node) {
+  std::string modulePath = node->scopeName();
+  size_t end = modulePath.size();
+  if (modulePath[end - 1] == '>') {
+    end = modulePath.rfind('<');
+  }
+  size_t start = modulePath.rfind('/', end);
+  start = (start != std::string::npos) ? start + 1 : 0;
+  return modulePath.substr(start, end - start);
+}
+
+c10::IValue getFunctionTuple(
+    const Module& module,
+    const Function& func,
+    bool save_debug_info_in_bytecode) {
   auto graph = func.graph()->copy();
+
   Inline(*graph);
-  torch::jit::Code code(graph, func.name());
+  if (save_debug_info_in_bytecode) {
+    ReconstructScopes(module, *graph, "top");
```

Comment:
I would probably prefer if we serialize the source of truth, i.e. the InlinedCallStacks, rather than a derivative from it, which is the scopes we reconstruct here.

Reply:
So, fundamentally I agree. This will also reduce the bloat in model size as we will be saving a tree rather than every visited path. My plan was to do this as a follow-up diff and let Tai continue his work as it may end up being broader in scope than what his internship allows.

Reply:
@iseeyuan, @ZolotukhinM are you guys ok with this plan for now? This will definitely blow up model size, but we should do follow-up diffs to manage that better.

Reply:
Looks fine for me for the debug info.